File: proverb.rs
pub fn build_proverb(list: &[&str]) -> String {
    let mut v = Vec::new();
    for i in 0..list.len(){
        match (list.get(i),list.get(i+1)){
            (Some(&a),Some(&b)) => {
                v.push(format!("For want of a {a} the {b} was lost."));
            },
            (Some(&_a),None) => {
                v.push(format!("And all for the want of a {0}.",list.get(0).unwrap()));
            }
            _ => println!("30/03/24")
        }
    }
    v.join("\n")
}

File: solutions/python/proverb/1/proverb.py
def proverb(*input_data, qualifier=None):
    if not input_data:
        return []
    lines = []
    for a, b in zip(input_data, input_data[1:]):
        lines.append(f"For want of a {a} the {b} was lost.")
    first = input_data[0]
    if qualifier:
        lines.append(f"And all for the want of a {qualifier} {first}.")
    else:
        lines.append(f"And all for the want of a {first}.")
    return lines